core/src/main/java/tech/pegasys/eth2signer/core/multikey/metadata/AbstractArtifactSignerFactory.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.eth2signer.core.multikey.metadata;
+
+import static java.lang.String.format;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import tech.pegasys.signers.azure.AzureKeyVault;
+import tech.pegasys.signers.hashicorp.HashicorpConnection;
+import tech.pegasys.signers.hashicorp.HashicorpConnectionFactory;
+import tech.pegasys.signers.hashicorp.TrustStoreType;
+import tech.pegasys.signers.hashicorp.config.ConnectionParameters;
+import tech.pegasys.signers.hashicorp.config.KeyDefinition;
+import tech.pegasys.signers.hashicorp.config.TlsOptions;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Optional;
+
+import com.google.common.io.Files;
+import org.apache.tuweni.bytes.Bytes;
+
+public abstract class AbstractArtifactSignerFactory implements ArtifactSignerFactory {
+
+  final HashicorpConnectionFactory connectionFactory;
+  final Path configsDirectory;
+
+  public AbstractArtifactSignerFactory(
+      final HashicorpConnectionFactory connectionFactory, final Path configsDirectory) {
+    this.connectionFactory = connectionFactory;
+    this.configsDirectory = configsDirectory;
+  }
+
+  protected Bytes extractBytesFromVault(final AzureSigningMetadata metadata) {
+    final AzureKeyVault azureVault =
+        new AzureKeyVault(
+            metadata.getClientId(),
+            metadata.getClientSecret(),
+            metadata.getTenantId(),
+            metadata.getVaultName());
+
+    final Optional<String> secret = azureVault.fetchSecret(metadata.getSecretName());
```

Comment:
nit: you may use `return azureVault.fetchSecret(metadata.getSecretName()).map(Bytes.from...).orElseThrow(...` to avoid using isEmpty and get methods.

Reply:
done